oak-commons/src/main/java/org/apache/jackrabbit/oak/commons/collections/CollectionUtils.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.jackrabbit.oak.commons.collections;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/**
+ * Utility methods for collections conversions.
+ */
+public class CollectionUtils {
+
+    private CollectionUtils() {
+        // no instances for you
+    }
+
+    /**
+     * Convert an iterable to a list.The returning list is mutable and supports all optional operations.
+     * @param iterable the iterable to convert
+     * @return the list
+     * @param <T> the type of the elements
+     */
+    public static <T> List<T> toList(final Iterable<T> iterable) {
+        List<T> result = new ArrayList<>();
+        iterable.forEach(result::add);
+        return result;
+    }
+
+    /**
+     * Convert an iterator to a list. The returning list is mutable and supports all optional operations.
+     * @param iterator the iterator to convert
+     * @return the list
+     * @param <T> the type of the elements
+     */
+    public static <T> List<T> toList(final Iterator<T> iterator) {
+        List<T> result = new ArrayList<>();
+        iterator.forEachRemaining(result::add);
+        return result;
+    }
+
+    /**
+     * Convert an iterable to a set. The returning set is mutable and supports all optional operations.
+     * @param iterable the iterable to convert
+     * @return the set
+     * @param <T> the type of the elements
+     */
+    public static <T> Set<T> toSet(final Iterable<T> iterable) {
```

Comment:
For consistency, could we implement this in an analogous way (forEachRemaining)?

Reply:
addressed in https://github.com/apache/jackrabbit-oak/pull/1679/commits/cf9b2bc93de33e61604d0559fd000c468d46c9f2